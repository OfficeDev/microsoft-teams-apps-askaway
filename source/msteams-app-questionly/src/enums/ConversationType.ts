/**
 * Define value for conversation type in teams app.
 */
export enum ConversationType {
    Channel = 'channel',
}
