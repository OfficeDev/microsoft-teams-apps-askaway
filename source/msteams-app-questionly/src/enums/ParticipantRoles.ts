/**
 * Define value for participant role in a meeting.
 */
export enum ParticipantRoles {
    Presenter = 'Presenter',
    Organizer = 'Organizer',
    Attendee = 'Attendee',
}
