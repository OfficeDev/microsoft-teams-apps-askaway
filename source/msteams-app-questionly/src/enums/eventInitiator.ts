/**
 * Define value for different event initiators.
 */
export enum EventInitiator {
    MainCard = 'MainCard',
    RestApi = 'RestApi',
}
