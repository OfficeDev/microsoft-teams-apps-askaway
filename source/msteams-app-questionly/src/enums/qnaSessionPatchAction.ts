/**
 * Patch actions supported for qna session.
 */
export enum QnaSessionPatchAction {
    End = 'end',
}
