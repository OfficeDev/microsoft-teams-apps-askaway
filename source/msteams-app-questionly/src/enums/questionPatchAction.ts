/**
 * Patch actions supported for question.
 */
export enum QuestionPatchAction {
    Upvote = 'upvote',
    Downvote = 'downvote',
    MarkAnswered = 'markAnswered',
}
