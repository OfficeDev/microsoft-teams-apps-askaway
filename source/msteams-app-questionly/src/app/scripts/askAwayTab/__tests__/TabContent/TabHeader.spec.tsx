/**
 * @jest-environment jsdom
 */

// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { AddIcon, RetryIcon } from '@fluentui/react-icons-northstar';
import { Button, Divider, Flex } from '@fluentui/react-northstar';
import { configure, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import enzymeToJson from 'enzyme-to-json';
import * as React from 'react';
import { ParticipantRoles } from '../../../../../enums/ParticipantRoles';
import { Helper } from '../../shared/Helper';
import { SwitchIcon } from '../../shared/Icons/SwitchIcon';
import TabHeader from '../../TabContent/TabHeader';

configure({ adapter: new enzymeAdapterReact16() });

describe('TabHeader Component', () => {
    let activeSessionData;
    let refreshSession;
    let t;
    let endSession;
    let showTaskModule;
    let onSwitchSessionClick;

    beforeAll(() => {
        t = jest.fn();
        activeSessionData = new Helper().createEmptyActiveSessionData();
        refreshSession = jest.fn();
        endSession = jest.fn();
        showTaskModule = jest.fn();
        onSwitchSessionClick = jest.fn();
    });

    it('should match the snapshot', () => {
        const userRole = ParticipantRoles.Presenter;
        const wrapper = shallow(
            <TabHeader
                t={t}
                refreshSession={refreshSession}
                endSession={endSession}
                showTaskModule={showTaskModule}
                activeSessionData={activeSessionData}
                userRole={userRole}
                disableActions={true}
                onSwitchSessionClick={onSwitchSessionClick}
            />
        );
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const userRole = ParticipantRoles.Presenter;
        const component = shallow(
            <TabHeader
                t={t}
                refreshSession={refreshSession}
                endSession={endSession}
                showTaskModule={showTaskModule}
                activeSessionData={activeSessionData}
                userRole={userRole}
                disableActions={true}
                onSwitchSessionClick={onSwitchSessionClick}
            />
        );

        expect(component.find(Flex)).toHaveLength(1);
        expect(component.find(Divider)).toHaveLength(1);
        expect(component.find(Button)).toHaveLength(3);
        expect(component.find(AddIcon)).toHaveLength(1);
        expect(component.find(RetryIcon)).toHaveLength(1);
        expect(component.find(SwitchIcon)).toHaveLength(1);
    });
});
