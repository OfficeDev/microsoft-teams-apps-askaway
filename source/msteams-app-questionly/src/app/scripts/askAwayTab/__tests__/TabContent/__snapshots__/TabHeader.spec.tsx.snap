// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabHeader Component should match the snapshot 1`] = `
<Fragment>
  <Flex
    className="tab-nav-header"
    gap="gap.small"
  >
    <Button
      accessibility={[Function]}
      as="button"
      disabled={true}
      onClick={[Function]}
      size="medium"
      text={true}
    >
      <RetryIcon
        xSpacing="after"
      />
      <ButtonContent
        as="span"
      />
    </Button>
    <Button
      accessibility={[Function]}
      as="button"
      disabled={true}
      onClick={[Function]}
      size="medium"
      text={true}
    >
      <AddIcon
        outline={true}
        xSpacing="after"
      />
      <ButtonContent
        as="span"
      />
    </Button>
    <Button
      accessibility={[Function]}
      as="button"
      disabled={true}
      onClick={[Function]}
      size="medium"
      text={true}
    >
      <SwitchIcon
        outline={true}
        xSpacing="after"
      />
      <ButtonContent
        as="span"
      />
    </Button>
  </Flex>
  <Divider
    size={0}
  />
</Fragment>
`;
